developer_manual/app/npm.rst

```diff
@@ -0,0 +1,76 @@
+===
+NPM
+===
+
+The node package manager ``npm`` is a widely used tool to manage JavaScript dependencies. In
+order to make the project setup easier for everybody, we recommend to follow the convention
+of typical scripts like ``build``, ``dev`` and ``test`` in the ``package.json`` file.
+
+You can find more info about this convention on blog posts like `this https://www.keithcirkel.co.uk/how-to-use-npm-as-a-build-tool/`_.
+
+npm build
+---------
+
+Once ``npm install`` (``npm i`` for short) has been run to install all dependencies specified
+on ``package.json``, one typically has to build the project. Building can happen with tools like
+Webpack, Grunt or similar. To unify the build command across Nextcoud apps, just add the build
+command(s) as ``build`` script to your.
+
+For apps that use webpack, this might look like this:
+
+.. code-block:: json
+
+	{
+	  "name": "myapp",
+	  "scripts": {
+		"build": "webpack --progress --config webpack.prod.js"
+	  },
+	  "devDependencies": {
+		"webpack": "^4.26.1",
+		"webpack-cli": "^3.1.2",
+	  }
+	}
+
+You can then run ``npm build`` in your app's root directory to invoke the build process.
+
+See the `npm-build docs https://docs.npmjs.com/cli/build`_ for more info.
+
+npm run dev
+-----------
+
+Since building the release version of JavaScript scripts can be slow, apps often have a dedicated
+build step for development that builds faster and enables debug output.
+
+This command should be added to ``package.json`` as ``dev`` script:
+
+.. code-block:: json
+
+	{
+	  "name": "myapp",
+	  "scripts": {
+		"build": "webpack --progress --config webpack.prod.js",
+        "dev": "webpack --progress --config webpack.dev.js"
```

Comment:
we're all using watch I think, shall we recommend that?

Reply:
yeah, unsure about that. watch is a dev build that watches (surprise!) changed files. why is there a `dev` build as well? what's that used for?

Reply:
have you ever used the `dev` script?

Reply:
I think just having the watch script is fine. If we add watch we should add a note that it probably requires increasing `fs.inotify.max_user_watches` which was an issue @violoncelloch had when trying to watch.

Reply:
> have you ever used the `dev` script?

Rarely, but yes when I need a single build and then do another command (and that I'm too lazy to open a new shell :grin: )